### PR TITLE
CMake: add check that *sanitizer builds require disabled ASLR

### PR DESCRIPTION
**Pull request by rschu1ze**
Sanitizer on systems with enabled ASLR lead to segfault at build time (llvm-tblgen) and runtime (clickhouse-server). This PR checks that ASLR is disabled.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit b1873aaacd1831f220fdc57b98ffa4ee3723fcac with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67005/b1873aaacd1831f220fdc57b98ffa4ee3723fcac/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/65521/9d7bb2a430b142e0761cc30efcf78ea1bbeb6871/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by alexey-milovidov — APPROVED** [review]

**Review by thevar1able — APPROVED** [review]

File: cmake/sanitize.cmake [hunk 1]
```diff
@@ -9,6 +9,16 @@ option (SANITIZE "Enable one of the code sanitizers" "")
 set (SAN_FLAGS "${SAN_FLAGS} -g -fno-omit-frame-pointer -DSANITIZER")
 
 if (SANITIZE)
```
Comment (nickitat) [inline cmake/sanitize.cmake:11]: afaiu, coverage doesn't fall into this category. IMO would make sense to check include it too

File: cmake/sanitize.cmake [hunk 2]
```diff
@@ -9,6 +9,16 @@ option (SANITIZE "Enable one of the code sanitizers" "")
...
+
+    if (OS_LINUX)
+        execute_process(COMMAND
+            sysctl --binary kernel.randomize_va_space
+            OUTPUT_VARIABLE RANDOMIZE_VA_SPACE)
+        if (NOT ${RANDOMIZE_VA_SPACE} STREQUAL "0")
+            message (FATAL_ERROR "Sanitizer builds require disabled address space layout randomization. Run 'sudo sysctl kernel.randomize_va_space=0' to disable it")
```
Comment (rschu1ze) [inline cmake/sanitize.cmake:18]: `sudo sysctl kernel.randomize_va_space=0` disables ASLR only until the next restart.

Permanent setting:
https://askubuntu.com/questions/318315/how-can-i-temporarily-disable-aslr-address-space-layout-randomization

**Comment by Algunenano** [comment]
Please keep in mind that https://github.com/ClickHouse/ClickHouse/pull/67081 superseedes this PR as it removes the need to disable ASLR.

**Comment by rschu1ze** [comment]
The two other options in https://github.com/ClickHouse/ClickHouse/issues/64086#issuecomment-2248852036 are more attractive --> closing.

**Pull request closed without merging**